src/tokens/tokens.service.ts

```diff
@@ -100,19 +108,34 @@ export class TokensService {
     this.shortPrefix = shortPrefix;
     this.username = username;
     this.password = password;
-    this.proxy.addListener(new TokenListener(this, this.topic));
+    this.proxy.addListener(new TokenListener(this));
   }
 
   private getMethodAbi(
     poolId: URLSearchParams,
-    operation: 'MINT' | 'TRANSFER' | 'BURN' | 'BALANCE' | 'TRANSFEREVENT',
+    operation: 'MINT' | 'TRANSFER' | 'BURN' | 'TRANSFEREVENT',
   ): IAbiMethod | undefined {
     const standard = poolId.get(EncodedPoolIdEnum.Standard) as ContractStandardStrings;
     const standardAbi: IAbiMethod[] = standardAbiMap[standard];
     const method = standardAbi?.find(abi => abi.name === standardMethodMap[standard][operation]);
     return method;
   }
 
+  private getAmountOrTokenID(
+    dto: TokenMint | TokenTransfer | TokenBurn,
+    type: TokenType,
+  ): string | undefined {
+    if (type === TokenType.FUNGIBLE) {
+      return dto.amount;
+    }
+
+    if (dto.amount !== undefined && dto.amount !== '1') {
+      throw new HttpException('Amount for nonfungible tokens must be 1', HttpStatus.BAD_REQUEST);
```

Comment:
You can use `BadRequestException` as a shorthand (amounts to the same thing though)